server/auth_function/test/lambda_function_test.py

```diff
@@ -126,12 +194,66 @@ def test_update_user_if_already_exists(
     assert count == 1
 
 
+@pytest.mark.parametrize(
+    "cognito_event",
+    [
+        "login_event_bceid.json",
+    ],
+    indirect=True,
+)
+def test_update_bceid_user_if_already_exists_test_with_business_guid(
+    db_pg_transaction,
+    cognito_event,
+    cognito_context,
+    test_user_properties,
+    initial_user_without_guid_or_cognito_id,
+):
+    """
+    if the user has already been created but does not have a guid or a cognito user
+    id then, it will be updated, and the guid / cognito user id will be populated.
+    """
+
+    test_idp_type_code = test_user_properties["idp_type_code"]
+    test_idp_user_id = test_user_properties["idp_user_id"]
+    test_cognito_user_id = test_user_properties["cognito_user_id"]
+    test_idp_username = test_user_properties["idp_username"]
+    test_idp_business_guid = test_user_properties["idp_business_id"]
+
+    # execute
+    result = lambda_function.lambda_handler(cognito_event, cognito_context)
+    LOGGER.debug(f"result: \n{pprint.pformat(result, indent=4)}")
+
+    # validate that there is one user in the database with the properties from
+    # the incoming event
+    cursor = db_pg_transaction.cursor()
+    raw_query = """select count(*) from app_fam.fam_user where
+        user_type_code = {} and
+        user_guid = {} and
+        cognito_user_id = {} and
+        user_name = {} and
+        business_guid = {};"""
+    query = sql.SQL(raw_query).format(
+        sql.Literal(test_idp_type_code),
+        sql.Literal(test_idp_user_id),
+        sql.Literal(test_cognito_user_id),
+        sql.Literal(test_idp_username),
+        sql.Literal(test_idp_business_guid),
```

Comment:
Test that for business bceid user login with attribute custom:idp_business_id configured readable and writable, if the user is already exists, will update the business_guid for the user, and we can get the user record back